pkgs/development/tools/misc/tideways/php-probe.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenvNoCC
+, lib
+, fetchurl
+, autoPatchelfHook
+, php
+}:
+
+let
+  soFile = {
+    "8.0" = "tideways-php-8.0.so";
+    "8.1" = "tideways-php-8.1.so";
+    "8.2" = "tideways-php-8.2.so";
+  }.${lib.versions.majorMinor php.version} or (throw "Unsupported PHP version.");
+in
+stdenvNoCC.mkDerivation rec {
```

Comment:
Do you think it would be possible implement the [finalAttrs pattern](https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes) here?